src/main/java/org/myrobotlab/service/interfaces/Simulator.java

```diff
@@ -14,13 +12,19 @@
  *
  */
 
+// FIXME !!! refactor this stuff out - create an AbstractSimulator ... 
 public interface Simulator extends NameProvider {
 
   // Generalized create ??? a good thing - router for ObjectFactory??
   // public Object create(ServiceInterface service);
 
+  /*
   public VirtualServo createVirtualServo(String name);
 
   public VirtualMotor createVirtualMotor(String name);
+  */
+  
+  ServoController getServoController();

```

Comment:
If this interface implements ServoController this method isn't needed.

Reply:
it is .. unless you can figure a way to support the gui and "attaching" correctly without it ..
I see this as a small cost compared to begin pushing all methods of all interfaces into the simulator :(